### PR TITLE
MCOL-4815 ColumnCommand was replaced with a set of derived classes sp…

### PR DESCRIPTION
**Pull request by drrtuy**
…ecified by

column width

RTSCommand was modified to use a fabric that produces CC class based on column width

NB this patch doesn't affect PseudoCC that also leverages ColumnCommand

### CODE REVIEW

File: primitives/primproc/columncommand.h [hunk 1]
```diff
@@ -175,6 +196,82 @@ class ColumnCommand : public Command
     friend class RTSCommand;
 };
 
+using ColumnCommandShPtr = std::unique_ptr<ColumnCommand>;
```
Comment (mariadb-AlexeyAntipovsky) [inline primitives/primproc/columncommand.h]: What does `Sh` in `ColumnCommandShPtr` stand for?

File: primitives/primproc/columncommand.cpp [hunk 2]
```diff
@@ -463,44 +544,20 @@ void ColumnCommand::processResult()
...
 }
 
+// Used by PseudoCC at least
 void ColumnCommand::createCommand(ByteStream& bs)
 {
     uint8_t tmp8;
-    uint32_t tmp32;
-
     bs.advance(1);
+    colType.unserialize(bs);
```
Comment (mariadb-AlexeyAntipovsky) [inline primitives/primproc/columncommand.cpp:552]: Could you please move Command serialization/deserialization closer to each other? Maybe it makes sense to create separate functions for this purpose, and in ColumnCommand and ColumnCommandJL just call them.

**Review by mariadb-AlexeyAntipovsky — CHANGES_REQUESTED** [review]

File: primitives/primproc/columncommand.h (hunk 1 above)
Comment (drrtuy) [reply primitives/primproc/columncommand.h]: Good point. The name is misleading.

File: primitives/primproc/columncommand.cpp (hunk 2 above)
Comment (mariadb-AlexeyAntipovsky) [reply primitives/primproc/columncommand.cpp:552]: Decided to do this refactoring separately.

**Review by mariadb-AlexeyAntipovsky — APPROVED** [review]

**Pull request merged**